src/Elastic.Transport/Extensions/SemVersion.cs

```diff
@@ -0,0 +1,73 @@
+// Licensed to Elasticsearch B.V under one or more agreements.
+// Elasticsearch B.V licenses this file to you under the Apache 2.0 License.
+// See the LICENSE file in the project root for more information
+
+using System.Diagnostics.CodeAnalysis;
+using System.Globalization;
+using System.Numerics;
+using System.Text.RegularExpressions;
+
+namespace Elastic.Transport.Extensions;
+
+internal sealed class SemVersion
+{
+	// https://semver.org/#is-there-a-suggested-regular-expression-regex-to-check-a-semver-string
+	private static readonly Regex Regex = new(@"^(0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*)(?:-((?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+([0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?$");
+
+	public BigInteger Major { get; }
```

Comment:
As we're now defining these properties ourselves, is there any reason not to prefer a regular integer in all cases? I can't see any situation in which we'd exceed the maximum value of that!

Reply:
Thanks for the review! I don't have a strong opinion on that. Might be good to have future proof code, if there are no significant drawbacks or complications. On the other hand, I agree that realistically we'll probably never exceed the integer space. Will change it to int 🙂